google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITDatabaseRolePermissionTest.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  @Test
+  public void grantAndRevokeDatabaseRolePermissions() throws Exception {
+    // Create database with table and role permission.
+    final String dbRoleParent = "parent";
+    final String databaseId = testHelper.getUniqueDatabaseId();
+    final String instanceId = testHelper.getInstanceId().getInstance();
+
+    final String createTableT = "CREATE TABLE T (\n" + "  K STRING(MAX),\n" + ") PRIMARY KEY(K)";
+    final String createRoleParent = String.format("CREATE ROLE %s", dbRoleParent);
+    final String grantSelectOnTableToParent =
+        String.format("GRANT SELECT ON TABLE T TO ROLE %s", dbRoleParent);
+
+    final Database createdDatabase =
+        dbAdminClient
+            .createDatabase(
+                instanceId,
+                databaseId,
+                ImmutableList.of(createTableT, createRoleParent, grantSelectOnTableToParent))
+            .get(5, TimeUnit.MINUTES);
+
+    // Connect to db with dbRoleParent.
+    SpannerOptions options = SpannerOptions.newBuilder().setCreatorRole(dbRoleParent).build();
+
+    Spanner spanner = options.getService();
+    DatabaseClient dbClient = spanner.getDatabaseClient(createdDatabase.getId());
+    DatabaseAdminClient dbAdminClient = spanner.getDatabaseAdminClient();
+
+    // Test SELECT permissions to role dbRoleParent on table T.
+    // Query using dbRoleParent should return result.
+    try (ResultSet rs =
+        dbClient.singleUse().executeQuery(Statement.of("SELECT COUNT(*) as cnt FROM T"))) {
+      assertTrue(rs.next());
+    } catch (PermissionDeniedException e) {
+      // This is not expected
+      fail("Got PermissionDeniedException when it should not have occurred.");
+    }
+
+    // Revoke select Permission for dbRoleParent.
+    final String revokeSelectOnTableFromParent =
+        String.format("REVOKE SELECT ON TABLE T FROM ROLE %s", dbRoleParent);
+
+    dbAdminClient
+        .updateDatabaseDdl(
+            instanceId, databaseId, Arrays.asList(revokeSelectOnTableFromParent), null)
+        .get();
+
+    // Test SELECT permissions to role dbRoleParent on table T.
+    // Query using dbRoleParent should return PermissionDeniedException.
+    try (ResultSet rs =
+        dbClient.singleUse().executeQuery(Statement.of("SELECT COUNT(*) as cnt FROM T"))) {
+      rs.next();
+      fail("Missing expected permission denied to dbRoleParent Exception");
+    } catch (SpannerException e) {
+      // This is what we expect.
+      assertThat(e.getErrorCode()).isEqualTo(ErrorCode.PERMISSION_DENIED);
+      assertThat(e.getMessage()).contains(dbRoleParent);
+    }
+    // Drop role and table.
+    final String dropTableT = "DROP TABLE T";
+    final String dropRoleParent = String.format("DROP ROLE %s", dbRoleParent);
+    dbAdminClient
+        .updateDatabaseDdl(instanceId, databaseId, Arrays.asList(dropTableT, dropRoleParent), null)
+        .get();
+  }
+
+  @Test
+  public void orphanRolePermissions() throws Exception {
```

Comment:
What exactly is this test case testing? That a role that has not received any specific permissions has no permissions? I think it would be good to add a comment to explain what it is that we are verifying here.